xla/stream_executor/stream_executor_internal.h

```diff
@@ -332,6 +332,8 @@ class StreamExecutorInterface {
   // Deallocates unified memory space previously allocated with
   // UnifiedMemoryAllocate.
   virtual void UnifiedMemoryDeallocate(void* mem) {}
+  virtual void* CollectiveMemoryAllocate(uint64_t size) = 0;
```

Comment:
These methods should be implemented at the host_gpu_executor:
https://github.com/openxla/xla/blob/main/xla/stream_executor/host/host_gpu_executor.h#L66

Otherwise HostExecutor will became abstract class which will lead to compilation error (allocating an object of abstract class type) here:
https://github.com/openxla/xla/blob/main/xla/stream_executor/host/host_platform.cc#L63